src/file.cc

```diff
@@ -1,19 +1,29 @@
 #include "tmfs.hh"
 
-int tmfs_read(const char * path, char * buf, size_t nbytes, off_t offset,
-              struct fuse_file_info * fi)
+int tmfs_open(const char * path, struct fuse_file_info * fi)
 {
   // get the real path
   std::string real_path = get_real_path(path);
 
   // open the file
-  int fd = open(real_path.c_str(), O_RDONLY);
+  int fd = open(real_path.c_str(), fi->flags);
```

Comment:
Reading through this, original code was better I think and more explicit:

```suggestion
  int fd = open(real_path.c_str(), O_RDONLY);
```